railties/lib/rails/generators/app_base.rb

```diff
@@ -197,6 +201,14 @@ def webserver_gemfile_entry # :doc:
         GemfileEntry.new("puma", "~> 3.11", comment)
       end
 
+      def purpose_gemfile_entry # :doc:
+        return [] unless options[:purposeful] == 'true'
+
+        generate_purpose_model
```

Comment:
this should probably be moved somewhere else (didn't get it to work at the moment)